tests/registries/test_handler_matching.py

```diff
@@ -30,72 +28,123 @@ def register_fn(registry, resource):
     raise Exception(f"Unsupported registry type: {registry}")
 
 
-@pytest.fixture()
-def cause_no_diff(resource):
-    return Mock(resource=resource, event='some-event', diff=None)
+@pytest.fixture(params=[
+    pytest.param(None, id='without-diff'),
+    pytest.param([], id='with-empty-diff'),
+])
+def cause_no_diff(request, resource):
+    return Mock(resource=resource, event='some-event', diff=request.param)
 
 
-@pytest.fixture()
+@pytest.fixture(params=[
+    pytest.param([('op', ('some-field',), 'old', 'new')], id='with-field-diff'),
```

Comment:
what does 'op' mean here?

Reply:
Operation. Usually it is one of: add, change, remove. In this case, it is irrelevant, as the object-under-test is the cause & handlers, not the diff — but a list of 4-item tuples is expected in the internal for-cycle, no matter what's inside, so we give it.